docs: Correct comment on toFQDN API definition

Alex-Waring opened the pull request:
This comment was added 5 years ago, and is no longer correct. Removing as discussed in slack conversation: https://cilium.slack.com/archives/C01JALNQAR1/p1692007243176749

This is a change in a comment to a file, so should need no release notes, or testing, just validation that the removal is correct.

joestringer reviewed (approved):
Thanks Alex, LGTM. This comment is for the old poller implementation which we haven't relied on for a long time. The proxy based ToFQDNs implementation doesn't have these limitations.

Alex-Waring committed:
docs: Correct comment on toFQDN API definition

The comment is 5 years old and no longer valid, removing and updating
the CRD.

Signed-off-by: Alex Waring <ajmwaring@gmail.com>
